### DISCUSSION TITLE
Label dropdown find-as-you-type matching too lenient

### DISCUSSION THREAD

**Discussion by pitrou** [community/community#40446] [Projects and Issues]
### Select Topic Area

Bug

### Body

Discussion title sums it up. At https://github.com/apache/arrow/ we have defined a bunch of [labels](https://github.com/apache/arrow/labels). The labels dropdown allows to start typing a part of the label and view select one of the matching results, which is great, but the matching is much too coarse. For example, if I start typing "enh", I get not only `Type: enhancement` amongst the results but also `Component: Archery`.

It seems that instead of doing a plain substring match, it's looking for every label that contains the typed characters regardless of where they appear.

(Note: I chose category "issues" but it also applies to PRs)

**Comment by ossanna16** [comment]
Hi there @pitrou and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)
